storage/innobase/buf/buf0buf.cc

```diff
@@ -4041,15 +4077,21 @@ dberr_t buf_page_t::read_complete(const fil_node_t &node)
              node.space->crypt_data &&
              node.space->crypt_data->type != CRYPT_SCHEME_UNENCRYPTED)
     {
-      ib::error() << "Cannot decrypt " << expected_id;
       err= DB_DECRYPTION_FAILED;
       goto release_page;
     }
     else
     {
-      ib::error() << "Space id and page no stored in the page, read in are "
-                  << read_id << ", should be " << expected_id;
-      err= DB_PAGE_CORRUPTED;
+      sql_print_error("InnoDB: Space id and page no stored in the page,"
+                      " read in from %s are "
+                      "[page id: space=" UINT32PF ", page number=" UINT32PF
+                      "], should be "
+                      "[page id: space=" UINT32PF ", page number=" UINT32PF
+                      "]",
+                      node.name,
+                      read_id.space(), read_id.page_no(),
+                      expected_id.space(), expected_id.page_no());
+      err= DB_FAIL;
       goto release_page;
```

Comment:
I see we are setting err to DB_FAIL here to indicate we have already reported the error and changing later to DB_PAGE_CORRUPTED as the function is no longer supposed to return DB_FAIL. A bit round about logic but should be fine.

Reply:
Right, with that we are signaling that we already reported a more specific error and do not want a generic error to be output:
```c++
release_page:
    if (recv_sys.free_corrupted_page(expected_id, node));
    else if (err == DB_FAIL)
      err= DB_PAGE_CORRUPTED;
    else
    {
      sql_print_error("InnoDB: Failed to read page " UINT32PF
                      " from file '%s': %s", expected_id.page_no(),
                      node.name, ut_strerr(err));
```
We must call `recv_sys.free_corrupted_page()` in every case, and I did not want to duplicate that call.